salt/modules/nagios_json.py

```diff
@@ -0,0 +1,199 @@
+        'service': [
+            'hostname',
+            'servicedescription',
+        ],
+        'host': [
+            'hostname',
+        ],
+    }
+    parameters['query'] = query
+    for param in query_params[query]:
+        parameters[param] = kwargs[param]
+
+    if not nagios_url.endswith('/'):
+        nagios_url = nagios_url + '/'
+
+    if 'cgi-bin' in nagios_url:
+        url = _urljoin(nagios_url, 'statusjson.cgi')
+    else:
+        url = _urljoin(nagios_url, 'cgi-bin/statusjson.cgi')
+
+    try:
+        if nagios_username and nagios_password:
+            result = requests.request(
+                method=method,
+                url=url,
+                headers=headers,
+                params=parameters,
+                data=data,
+                verify=True,
+                auth=(nagios_username, nagios_password)
+            )
+        else:
+            result = requests.request(
+                method=method,
+                url=url,
+                headers=headers,
+                params=parameters,
+                data=data,
+                verify=True,
+            )
+        if result.status_code == 200:
+            data = result.json()
+        elif result.status_code == 401:
+            log.info('Authentication failed. Check nagios_username and nagios_password.')
+        elif result.status_code == 404:
+            log.info('Url {0} not found.'.format(url))
+        else:
+            log.info('Results: {0}'.format(result.text))
+    except ConnectionError as _error:
+        log.info('Error {0}'.format(_error))
+    return data
+
+
+def service_status(hostname, service_description):
+    '''
+    Check the status in Nagios for a particular
+    service on a particular host
+
+    :param hostname:                The hostname to check the status of the service in Nagios.
+    :param service_description:     The service to check the status of in Nagios.
+    :return: Boolean                True is the status is 'OK' or 'Warning', False if 'Critical'
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' nagios_json.service_status hostname=webserver.domain.com service_description='HTTP'
+
+    '''
+
+    config = _config()
+
+    if not config['nagios_url']:
+        log.error('Missing nagios_url')
+        return False
+
+    results = _status_query(query='service',
+                            nagios_url=config['nagios_url'],
+                            nagios_username=config['nagios_username'],
+                            nagios_password=config['nagios_password'],
+                            hostname=hostname,
+                            servicedescription=service_description)
+
+    data = results.get('data', '')
+    if data:
+        status = data.get('service', '').get('status', '')
+        if status and status == 0:
+            return False
+        elif status and status > 0:
+            return True
+        else:
+            return False
+    else:
+        return False
```

Comment:
Here is something very-very strange to me.

First there is a clearly a bug: `status` cannot be an empty string **and** 0! Another bug is that if dict `data` has no key "service", you return an empty string, which has no method `get`, so you won't get your 'status' but crash instead!

Second, you do not need these two "else" at all. I would get rid of this puzzle and write it simpler (assuming data is `None` here):
```python
return (data or {}).get('service', {}).get('status', 0) > 0
```
That's all you need.